lib/services/elasticsearch.js

```diff
@@ -1597,94 +1540,143 @@ class ElasticSearch extends Service {
       }
     }
   }
-}
-
-module.exports = ElasticSearch;
 
-/**
- * Builds a request formatted for Elasticsearch service
- * and map the name 'collection' to 'type' for ES
- *
- * @param {Request} request
- * @param {Array} [extraParams] [optional] An array of String values corresponding
- *                            to the extra params from the Kuzzle Request to be
- *                            included in the Elasticsearch Request.
- * @return {object} data the data with cleaned attributes
- */
-function initESRequest(request, extraParams = []) {
-  const
-    data = {},
-    index = getIndex(request);
+  _extractIndexes (esIndexes, { internal=false } = {}) {
+    const
+      indexes = new Set(),
+      prefix = internal
+        ? INTERNAL_PREFIX
+        : USER_PREFIX;
+
+    for (const esIndex of esIndexes) {
+      if (esIndex[0] === prefix) {
+        indexes.add(extractIndex(esIndex));
+      }
+    }
 
-  if (index) {
-    data.index = index;
+    return Array.from(indexes);
   }
 
-  if (request.input.resource.collection) {
-    data.type = request.input.resource.collection;
+  /**
+   * Returns a new elasticsearch client instance
+   *
+   * @returns {object}
+   */
+  _buildClient () {
+    // Passed to Elasticsearch's client to make it use
+    // Bluebird instead of ES6 promises
+    const defer = function defer () {
+      let
+        resolve,
+        reject;
+
+      const promise = new Bluebird((res, rej) => {
+        resolve = res;
+        reject = rej;
+      });
+
+      return { resolve, reject, promise };
+    };
+
+    return new ESClient(Object.assign({ defer }, this.config.client));
   }
 
-  for (const name of extraParams) {
-    if (typeof request.input.args[name] !== 'undefined') {
-      data[name] = request.input.args[name];
+  /**
+   * Get the name of an esIndex for an index + collection
+   *
+   * @param {string} index
+   * @param {string} collection
+   *
+   * @returns {string} esIndex name (eg: '&nepali#liia')
+   */
+  _getESIndex (index, collection) {
+    if (! index || ! collection) {
+      return null;
     }
+
+    const prefix = this.internal
+      ? INTERNAL_PREFIX
+      : USER_PREFIX;
+
+    return `${prefix}${index}${NAME_SEPARATOR}${collection}`;
   }
 
-  return data;
-}
+  /**
+   * Remove depth in object (replace hits.hits<array>, with hits<array>)
+   *
+   * @param result
+   * @returns {object}
+   */
+  _formatSearchResults (result) {
+    const formattedResult = result.hits
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "formattedResult". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)